### PR TITLE
feat: Add support for default_route_settings

### LINKED ISSUE #30: be able to control detailed_metrics_enabled, throttling_burst_limit, throttling_rate_limit
## Is your request related to a new offering from AWS?
no

## Is your request related to a problem? Please describe.
missing exposed functionality

## Describe the solution you'd like.
be able to control: detailed_metrics_enabled, throttling_burst_limit, throttling_rate_limit

## Describe alternatives you've considered.
coming from flat resources, want to replace them with this module.

## Additional context
```
   ~ default_route_settings {
          ~ detailed_metrics_enabled = true -> false
          - throttling_burst_limit   = 100 -> null
          - throttling_rate_limit    = 100 -> null
            # (1 unchanged attribute hidden)
        }
        # (1 unchanged block hidden)
```

I actually see code for this https://github.com/terraform-aws-modules/terraform-aws-apigateway-v2/blob/5a7f67401c96bf6856fae8867da4492d9ac5af84/main.tf#L67 - but for some reason it's commented out?

it's problematic to not be able to set the limits as they will then be null, and all traffic will get throttled/denied.

### PR DESCRIPTION
**Pull request by davidkarlsen**
## Description
Add support for default_route_settings, fixes #30.

## Motivation and Context
Control throttling and metrics.

## Breaking Changes
No

## How Has This Been Tested?
- [X] I have tested and validated these changes using one or more of the provided `examples/*` projects

Examples updated, also applied on our own real world infra.

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -65,6 +65,12 @@ variable "description" {
   default     = null
 }
 
+variable "default_route_settings" {
+  description = "Settings for default route"
+  type        = map(string)
+  default     = null
```
Comment (antonbabenko) [inline variables.tf:71]:
```suggestion
  default     = {}
```

**Review by antonbabenko — COMMENTED** [review]
Looks good, minor comment.

**Comment by davidkarlsen** [comment]
@antonbabenko fixed, what's up with the docs?

**Comment by antonbabenko** [comment]
Thanks again!

I rerun `pre-commit run -a` locally to fix the docs (c536a97).

[v0.16.0](https://github.com/terraform-aws-modules/terraform-aws-apigateway-v2/releases/v0.16.0) has been just released.

**Pull request merged**